validator/valnode/validation_api.go

```diff
@@ -148,6 +148,19 @@ func (a *ExecServerAPI) GetStepAt(ctx context.Context, execid uint64, position u
 	return server_api.MachineStepResultToJson(res), nil
 }
 
+func (a *ExecServerAPI) GetMachineHashesWithStepSize(ctx context.Context, execid, fromStep, stepSize, maxIterations uint64) ([]common.Hash, error) {
+	run, err := a.getRun(execid)
+	if err != nil {
+		return nil, err
+	}
+	leavesInRange := run.GetMachineHashesWithStepSize(fromStep, stepSize, maxIterations)
```

Comment:
*maybe* name this variable `hashesInRange`
The `leaves` thing seems to come out of nowhere.